beets/dbcore/types.py

```diff
@@ -125,7 +131,7 @@ class Integer(Type):
     query = query.NumericQuery
     model_type = int
 
-    def normalize(self, value):
+    def normalize(self, value: str) -> Union[int, str]:
```

Comment:
We could define a type alias per class?

Reply:
Seems like the solution is to make `Type` generic over `model_type` and `null_type` cf. https://stackoverflow.com/questions/74885777/how-to-implement-associated-types-in-python-mypy-or-what-to-do-when-wanting-su

How much time would you still like to invest into this project? If you like, go ahead, otherwise, I'd also offer to merge this quickly, and help with improving it myself afterwards.

Reply:
I'm happy to invest more time into this but maybe in another PR? You seem to have a good idea of what you'd like the system to be for these classes' typing, and it would also mean I can work on the pathlib PR more easily.